viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
+    # make copy of user_data watched to use as list to remove movies also watched by friends
+    unique_movies = copy(user_data["watched"])
+
+    for my_movie in user_data["watched"]: #access list in user watched
+        for friend in user_data["friends"]: #access friend lists in user friends
+            for friend_movie in friend["watched"]: #access movie list in each friend watched
+                if my_movie["title"] == friend_movie["title"] and my_movie in unique_movies:
+                    unique_movies.remove(my_movie)
+
+    return unique_movies
+    
+ 
+def get_friends_unique_watched(user_data):
+    '''Returning list of unique movies watched by friends and not user'''
+    friend_movies = []
+
+    #create combined friend movie list
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friend_movies:
+                friend_movies.append(movie)
+    
+    #compare user movies to friend movies    
+    for my_movie in user_data["watched"]:
+        for movie in friend_movies:
+            if movie["title"] == my_movie["title"]:
+                friend_movies.remove(movie)
+
+    return friend_movies
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    '''Creating friend movie recommendation list for user. User must have subscription
+    service where movie is streamed and not seen the movies yet'''
+
+    recommended_movies = []
+    
+    # get list of friend movies not in user watched list
+    friend_movies = get_friends_unique_watched(user_data)
+
+    # check if user subscriptions includes friend movie host
+    for movie in friend_movies:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+    
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    '''Creating friend recommended movie list if user has not seen it and
+    genre matches user most frequently watched genre'''
+    recommended_movies = []
+    user_most_watched_genre = get_most_watched_genre(user_data)
+    friend_unique_movies = get_friends_unique_watched(user_data)
+
+    for movie in friend_unique_movies:
+        if movie["genre"] == user_most_watched_genre:
+            recommended_movies.append(movie)
+        
+    return recommended_movies
+
+
+def get_rec_from_favorites(user_data):
+    '''Creating user recommended movie list for friends. User movies are favorites and have
+    not been watched by friends'''
+    recommended_movies =[]
+    friend_watched_movies = []
+    
+    # Get list of movies watched by all friends
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_watched_movies.append(movie)

```

Comment:
It sounds like you're looking for all of the movies that the user has watched that their friends have not watched. That sounds a lot like what the `get_unique_watched` function does. These lines of code can be replaced by calling that function. Then, in line 166, instead of `movie not in friend_watched_movies` it can be `movie in unique_watched_list`.